modules/ROOT/pages/indexes-for-nearest-neighbor.adoc

```diff
@@ -0,0 +1,431 @@
+== Drop vector indexes
+
+A vector index is dropped by using the xref:indexes-for-search-performance.adoc#indexes-drop-an-index[same command as for other indexes], `DROP INDEX`.
+
+.+DROP INDEX+
+======
+
+In the following example, you drop the `abstract-embeddings` that you created previously:
+
+.Query
+[source,cypher]
+----
+DROP INDEX `abstract-embeddings`
+----
+
+.Result
+[queryresult]
+----
+Removed 1 index.
+----
+
+======
+
+[[indexes-vector-set]]
+== Set vector property on a node
+
+You can set a vector property, `LIST<FLOAT>`, on a node using the xref:clauses/set.adoc#set-set-a-property[`SET`] command:
+
+.Syntax for setting a vector property
+[source,cypher]
+----
+SET node.propertyKey = vector
+----
+
+However, Cypher stores the provided `LIST<FLOAT>` as a primitive array of {ieee-754} _double_ precision values.
+Valid vectors for use in the index must have components finitely representable in {ieee-754} _single_ precision.
+As a result, it takes up approximately twice as much space.
+
+The {link-procedures-reference}#procedure_db_create_setVectorProperty[`db.create.setVectorProperty`] procedure is provided to validate and set the property as an array of {ieee-754} single precision values.
+As a result, the store's space is approximately halved.
+
+.Signature for `db.create.setVectorProperty` to set a vector property
+[source, syntax, role="noheader", indent=0]
+----
+db.create.setVectorProperty(node :: NODE?, key :: STRING?, vector :: LIST? OF FLOAT?) :: (node :: NODE?)
+----
+
+Thus, the equivalent syntax for setting as the aforementioned:
+
+.Syntax for setting a vector _via_ a procedure
+[source,cypher]
+----
+CALL db.create.setVectorProperty(node, 'propertyKey', vector)
+----
+
+This is ideal for creating and setting new vector properties in the graph.
+Though existing properties can be re-set using this procedure to minimize store size, there is a cost in transaction log size until they are rotated away, especially in the case of record formats.
+
+[[indexes-vector-similarity]]
+== Supported similarity functions
+
+The choice of similarity function can vastly differ in which vectors are considered similar, and which are valid.
+The semantic meaning of the vector may itself dictate which similarity function to choose.
+Consult the documentation for the particular model used to generate vector embeddings.
+The documentation may suggest a preference for certain similarity functions.
+Otherwise, understanding the differences can help make an informed choice.
+
+.Similarity functions
+[%header,cols="d,m,e"]
+|===
+| Name | Case Insensitive Argument | Key similarity feature
+
+| xref:#indexes-vector-similarity-euclidean[Euclidean]
+| "euclidean"
+| distance
+
+| xref:indexes-vector-similarity-cosine[Cosine]
+| "cosine"
+| angle
+
+|===
+
+For {l2-norm}alized vectors, such that their length, image:l2norm_is_1.svg["The l2-norm of vector v equals 1"], both Euclidean and cosine similarity functions produce the same similarity ordering.
```

Comment:
Is this still accurate?
```suggestion
For {l2-norm}alized vectors, where the length is image:l2norm_is_1.svg["The l2-norm of vector v equals 1"], both Euclidean and cosine similarity functions produce the same similarity ordering.
```

Reply:
I am not sure about the use of "where" here. It _can_ make sense, but could also be read to indicate that there could be l2-normalised vectors where the length is not 1 … which is impossible.

What are your thoughts about:
```suggestion
For {l2-norm}alized vectors (unit vectors), thus their length image:l2norm_is_1.svg["The l2-norm of vector v equals 1"], both Euclidean and cosine similarity functions produce the same similarity ordering.
```